### PR TITLE
Add securedrop-proxy logging

### LINKED ISSUE #44: Log securedrop-proxy exceptions
A requirement for freedomofpress/securedrop-workstation#19 and something that would generally be useful for developers is ensuring that at least exceptions are logged to a file for debugging purposes

### PR DESCRIPTION
**Pull request by creviera**
Resolves #44 

## Test

1. Make a `securedrop-proxy` debian package from this branch
2. Copy it to `sd-proxy` VM (you might need to add a FileCopy policy to allow this
3. `dpkg` install it
4. Run the client
5. Tail `~/.securedrop-proxy/logs/proxy.log` in `sd-proxy` and see your requests going through

I was able to cause a `BrokenPipeError` and see the traceback in the logs. The proxy was able to gracefully recover on the next request, but we now have record of things like that happening.

### CODE REVIEW

File: securedrop_proxy/entrypoint.py [hunk 1]
```diff
@@ -6,18 +6,38 @@
...
+    Set up a new proxy object with an error handler, configuration that we read from  argv[1], and
+    the original user request from STDIN.
+    '''
+    try:
+        configure_logging()
+    except Exception as e:
+        print(e)
+        return
+
+    logging.info('Starting SecureDrop Proxy {}'.format(version))
```
Comment (emkll) [inline securedrop_proxy/entrypoint.py]: Perhaps setting to debug here (and other similar messages that don't provide any request-specific information) would make sense?

**Comment by creviera** [comment]
> What do you think about setting the logging threshold to INFO? 

Yes to this suggestion! I didn't realize it defaulted to DEBUG level. I'll make the change.

**Comment by creviera** [comment]
nevermind it didn't default, i set it to debug level because i copy-pasted from the client... this should be updated in the client too

File: securedrop_proxy/proxy.py [hunk 2]
```diff
@@ -134,11 +139,13 @@ def proxy(self):
                 raise ValueError("Request callback is not set.")
```
[the remark this answers is not in the document]
Comment (creviera) [reply securedrop_proxy/proxy.py]: Caught and logged elsewhere. If you look down a bit to lines 147-148, you'll see that ValueError is caught and logged there.

File: securedrop_proxy/proxy.py [hunk 3]
```diff
@@ -99,7 +103,7 @@ def handle_non_json_response(self):
         res = Response(self._presp.status_code)
 
         # Create a NamedTemporaryFile, we don't want
-        # to delete it after closing.
+        # to delete it after closign.
```
[the remark this answers is not in the document]
Comment (creviera) [reply securedrop_proxy/proxy.py]: yes, i did not intend to add a typo to a random comment :P i'll change it back

File: securedrop_proxy/proxy.py [hunk 4]
```diff
@@ -71,7 +75,7 @@ def prep_request(self):
         try:
```
[the remark this answers is not in the document]
Comment (creviera) [reply securedrop_proxy/proxy.py]: > https://github.com/freedomofpress/securedrop-proxy/pull/47/files#diff-d6a239b06b6291805f2c69f02ac4abaeR73

so `prep_request` is called inside a try-catch block that logs the error

> https://github.com/freedomofpress/securedrop-proxy/pull/47/files#diff-d6a239b06b6291805f2c69f02ac4abaeR55

i'm confused, you linked to the method definition line, where should logging be?

File: securedrop_proxy/entrypoint.py [hunk 5]
```diff
@@ -44,3 +64,42 @@ def start():
...
+    This function is called in the event of a catastrophic failure.
+    Log exception and exit cleanly.
+    '''
+    logging.error('Unrecoverable error', exc_info=(exc_args))
+    sys.__excepthook__(*exc_args)
+    print('')  # force terminal prompt on to a new line
+    sys.exit(1)
+
+
+def configure_logging() -> None:
```
[the remark this answers is not in the document]
Comment (creviera) [reply securedrop_proxy/entrypoint.py]: good idea

**Comment by creviera** [comment]
hey @emkll - you should no longer see any debug logging now when you follow the log file, unless you set the env var for loglevel to be debug

ready for review

**Pull request merged**